qiskit/visualization/pulse_v2/generators/waveform.py

```diff
@@ -105,10 +105,11 @@ def gen_filled_waveform_stepwise(
             if isinstance(pval, circuit.ParameterExpression):
                 unbound_params.append(pname)
 
-        if hasattr(data.inst.pulse, "pulse_type"):
-            pulse_shape = data.inst.pulse.pulse_type
+        pulse_data = data.inst.pulse
+        if isinstance(pulse_data, library.SymbolicPulse):
+            pulse_shape = pulse_data.pulse_type
         else:
-            pulse_shape = data.inst.pulse.__class__.__name__
+            pulse_shape = "Waveform"
```

Comment:
I'm not super sure what this magic value represents, but it appears a couple of times.  Should it be in the other enum of magic values that appeared earlier?